realis.go

```diff
@@ -504,11 +504,14 @@ func (r *realisClient) ReestablishConn() error {
 	// Close existing connection
 	r.logger.Println("Re-establishing Connection to Aurora")
 
+	// This call must happen before we lock as it also uses
+	// the same lock from the client since close can be called
+	// by anyone from anywhere.
+	r.Close()
```

Comment:
Why does Close need a lock?

It seems the transport is set in the New function and never changes. Is there another code path that may change it?

Reply:
This was an effort in trying to chase down the bug this patch fixed: https://github.com/apache/thrift/commit/aa9e7e8f5966ec9233ef418bab81b48c5a539028

After finding the source of that issue, you may be right that this critical section may be a candidate for removal.

In theory, this close method is public and can be called by multiple threads all at once. I'll have to check if closing a connection multiple times results in an error with the thrift library.

Reply:
Makes sense, if the transport close method is not safe for concurrent calls